armi/reactor/converters/axialExpansionChanger/assemblyAxialLinkage.py

```diff
@@ -12,16 +12,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import typing
+import dataclasses
+import functools
+import itertools
+
 from armi import runLog
-from armi.reactor.components import UnshapedComponent
+from armi.reactor.blocks import Block
+from armi.reactor.components import Component, UnshapedComponent
 from armi.reactor.converters.axialExpansionChanger.expansionData import (
-    getSolidComponents,
+    iterSolidComponents,
 )
 
 
-def _determineLinked(componentA, componentB):
+if typing.TYPE_CHECKING:
+    from armi.reactor.assemblies import Assembly
+
+
+def areAxiallyLinked(componentA: Component, componentB: Component) -> bool:
     """Determine axial component linkage for two components.
 
+    Components are considered linked if the following are found to be true:
+
+    1. Both contain solid materials.
+    2. They have compatible types (e.g., ``Circle`` and ``Circle``).
+    3. Their multiplicities are the same.
+    4. The smallest inner bounding diameter of the two is less than the largest outer
```

Comment:
```suggestion
    4. The biggest inner bounding diameter of the two is less than the smallest outer
```
For the change, consider a pin above an annulus where the pin would fit into the annulus (so not overlapping). 
- The smallest inner bounding diameter is the pin (it is zero)
- zero is less than the OD of the annulus; so by that logic, they would overlap, which they don't. 

But if we flip it, 
- The biggest ID is that of the annulus. 
- If the ID of the annulus is less than the smaller of the ODs (the pin), then we're good.

Reply:
Okay that tracks and matches the code